### PR TITLE
Make sure rsync.sh is executable inside the build image

### PR DESCRIPTION
**Pull request by ixdy**
I kept having the build fail:

``` console
$ make quick-release
+++ [1006 18:13:44] Verifying Prerequisites....
+++ [1006 18:13:44] Building Docker image kube-build:build-d3c60cf83f-3-v1.6.3-9
+++ [1006 18:13:54] Creating data container kube-build-data-d3c60cf83f-3-v1.6.3-9
+++ [1006 18:13:55] Syncing sources to container
!!! [1006 18:16:01] Could not connect to rsync container. See build/README.md for setting up remote Docker engine.
make: *** [quick-release] Error 1
```

`docker ps` revealed the issue:

``` console
$ docker ps
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
$ docker ps -a
CONTAINER ID        IMAGE                                    COMMAND                  CREATED             STATUS                       PORTS               NAMES
75c2a3c40cb3        kube-build:build-d3c60cf83f-3-v1.6.3-9   "/rsyncd.sh"             6 seconds ago       Exited (126) 5 seconds ago                       kube-rsync-d3c60cf83f-3-v1.6.3-9
3eb215e41f36        kube-build:build-d3c60cf83f-3-v1.6.3-9   "chown -R 85078.5000 "   8 seconds ago       Exited (0) 6 seconds ago                         kube-build-data-d3c60cf83f-3-v1.6.3-9
5a2707af2ccd        882577c54f67                             "/bin/sh -c 'cd ${K8S"   7 days ago          Exited (2) 7 days ago                            stupefied_goldberg
$ docker logs 75c2a3c40cb3
/bin/bash: /rsyncd.sh: Permission denied
```

I'm not sure why this works on Jenkins but not on my machine.

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/34288)

### CODE REVIEW

**Comment by ixdy** [comment]
I also moved `build/BUILD_IMAGE_VERSION` to `build/image/VERSION`, since the cross-build image uses `build/build-image/cross/VERSION`, but I can change this back if you feel strongly.

File: build/build-image/Dockerfile [hunk 1]
```diff
@@ -37,3 +37,4 @@ ENV KUBE_GIT_VERSION_FILE ${HOME}/.dockerized-kube-version-defs
 ADD rsyncd.password /
 RUN chmod a+r /rsyncd.password
 ADD rsyncd.sh /
+run chmod a+rx /rsyncd.sh
```
Comment (jbeda) [inline build/build-image/Dockerfile]: run should be RUN

**Review by jbeda — APPROVED** [review]
Just the capitalization nit. Otherwise LGTM. Feel free to self apply label as I'm mobile.

File: build/build-image/Dockerfile (hunk 1 above)
Comment (ixdy) [inline build/build-image/Dockerfile]: not sure how I even did that. thanks for catching this. :)

**Comment by ixdy** [comment]
Updated this PR to fix `/etc/localtime`, too, since that is also affected by a restrictive umask (which appears to be the root cause).

**Comment by colemickens** [comment]
I cherry-picked this after rebasing on top of this fix: https://github.com/kubernetes/kubernetes/pull/34257

I'm still having issues. Not sure if it's supposed to be fixed by this PR. Not sure where is the best place to dig in for debugging yet:
```shell
+++ [1006 19:38:24] Verifying Prerequisites....
+++ [1006 19:38:28] Removing _output directory
+++ [1006 19:38:28] Verifying Prerequisites....
+++ [1006 19:38:28] Building Docker image kube-build:build-be4685d15b-4-v1.6.3-9
+++ [1006 19:39:08] Creating data container kube-build-data-be4685d15b-4-v1.6.3-9
+++ [1006 19:39:09] Syncing sources to container
!!! [1006 19:39:13] Could not connect to rsync container. See build/README.md for setting up remote Docker engine.
make: *** [Makefile:239: release] Error 1
```

**Comment by jbeda** [comment]
Cole -- this sounds like a different issue. Can you open a new issue with
details of your setup?

**Comment by ixdy** [comment]
@jbeda can I get an LGTM?

**Comment by jbeda** [comment]
Sorry for the delay!  Had some meetings this morning and thought we had it figured out last night.

**Pull request merged**